internal/workflows/workflows.go

```diff
@@ -0,0 +1,168 @@
+		PodMetadata: &argo.Metadata{
+			Labels: map[string]string{
+				"ame-task": t.GetName(),
+			},
+		},
+		Templates: []argo.Template{
+			{
+				Steps: wfSteps,
+				Name:  "main",
+			},
+		},
+		Entrypoint: "main",
+		VolumeClaimTemplates: []apiv1.PersistentVolumeClaim{
+			{
+				ObjectMeta: v1.ObjectMeta{
+					Name: taskVolName(t),
+				},
+				Spec: apiv1.PersistentVolumeClaimSpec{
+					AccessModes: []apiv1.PersistentVolumeAccessMode{
+						apiv1.ReadWriteOnce,
+					},
+					Resources: apiv1.ResourceRequirements{
+						Requests: apiv1.ResourceList{
+							"storage": volStoreageReq,
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func taskVolName(t *v1alpha1.Task) string {
+	return fmt.Sprintf("%s-volume", t.GetName())
+}
+
+func genTaskSpecFromPipelineStep(parentTask *v1alpha1.Task, step v1alpha1.PipelineStep) *v1alpha1.TaskSpec {
+	spec := parentTask.Spec.DeepCopy()
+	spec.RunCommand = step.RunCommand
+	spec.Pipeline = nil
+	spec.Env = step.Env
+	spec.Secrets = step.Secrets
+	return spec
+}
+
+func genTemplate(templateName string, t *v1alpha1.Task, volName string) argo.Template {
+	return argo.Template{
+		Name: templateName,
+
+		Script: &argo.ScriptTemplate{
+			Source: fmt.Sprintf(`
+
+          export TASK_DIRECTORY=ameprojectstorage/%s
+
+          s3cmd --no-ssl --region us-east-1 --host=$MINIO_URL --host-bucket=$MINIO_URL get --recursive s3://$TASK_DIRECTORY ./
+
+          cd "./%s" 
+
+          set -e # It is important that the workflow exits with an error code if execute or save_artifacts fails, so AME can take action based on that information.
+
+          execute "%s" 
+          
+          save_artifacts "ameprojectstorage/%s/artifacts/"
+
+          echo "0" >> exit.status
+					`, t.Spec.ProjectId, t.Spec.ProjectId, t.Spec.RunCommand, t.GetName()),
+			Container: apiv1.Container{
+				Image: "ame-executor:local",
+				Command: []string{
+					"bash",
+				},
+				Env: append([]apiv1.EnvVar{
+					{
+						Name:  "AWS_ACCESS_KEY_ID",
+						Value: "minio",
+					},
+					{
+						Name:  "AWS_SECRET_ACCESS_KEY",
+						Value: "minio123",
+					},
+					{
+						Name:  "MINIO_URL",
+						Value: "http://ame-minio.ame-system.svc.cluster.local:9000",
+					},
+					{
+						Name:  "PIPENV_YES",
+						Value: "1",
+					},
+				}, taskEnvToContainerEnv(t.Spec)...),
+				VolumeMounts: []apiv1.VolumeMount{
+					{
+						Name:      volName,
+						MountPath: "/project",
+					},
+				},
+			},
+		},
+
+		PodSpecPatch: `{"containers":[{"name":"main", "resources":{"limits":{
+        "memory": "3Gi"   }}}]}`,
+	}
+}
+
+func taskEnvToContainerEnv(t v1alpha1.TaskSpec) []apiv1.EnvVar {
+	var v1env []apiv1.EnvVar
+	for _, e := range t.Env {
+		v1env = append(v1env, apiv1.EnvVar{
+			Name:  e.Name,
+			Value: e.Value,
+		})
+	}
+
+	for _, s := range t.Secrets {
+		v1env = append(v1env, apiv1.EnvVar{
+			Name: s.EnvKey,
+			ValueFrom: &apiv1.EnvVarSource{
+				SecretKeyRef: &apiv1.SecretKeySelector{
+					Key: "secret",
```

Comment:
I think we should move the key used by AME when creating and reading from secrets to separate variable which can be used across the project.